### PR TITLE
Add test cases for prometheus component

### PR DESCRIPTION
**Pull request by rahulsrivastava71**
Provide a summary of the change and which issue (i.e. ticket) is fixed.
If there are any dependencies, for example on a PR in another repository, list those.

### CODE REVIEW

File: platform-operator/controllers/verrazzano/component/prometheus/adapter/promadapter_component_test.go [hunk 1]
```diff
@@ -71,3 +72,77 @@ func TestIsEnabled(t *testing.T) {
...
+							InstallOverrides: vzapi.InstallOverrides{MonitorChanges: &trueValue},
+						},
+					},
+				},
+			},
+			expectTrue: true,
+		},
+		{
+			// GIVEN a Verrazzano custom resource with the Prometheus Adapter disabled
+			// WHEN we call IsReady on the Prometheus Adapter component
```
Comment (anajosep) [inline platform-operator/controllers/verrazzano/component/prometheus/adapter/promadapter_component_test.go:119]: Test method is `MonitorOverrides` Can you correct the comments.

File: platform-operator/controllers/verrazzano/component/prometheus/adapter/promadapter_component_test.go [hunk 2]
```diff
@@ -71,3 +72,77 @@ func TestIsEnabled(t *testing.T) {
 		})
 	}
 }
+
+// Test isReady for not nil when it's called with component context
+func TestIsReady(t *testing.T) {
+	c := fake.NewClientBuilder().Build()
+	ctx := spi.NewFakeContext(c, &vzapi.Verrazzano{}, nil, false, profilesRelativePath)
+	assert.NotNil(t, NewComponent().IsReady(ctx))
```
Comment (anajosep) [inline platform-operator/controllers/verrazzano/component/prometheus/adapter/promadapter_component_test.go:80]: Can you test different combinations, Like when dry run flag is set to true and the helm component's `IsReady` call returns true. Assertion should be meaningful and checked for `true` or `false` (and not for `nil`)

File: platform-operator/controllers/verrazzano/component/prometheus/adapter/promadapter_component_test.go [hunk 3]
```diff
@@ -71,3 +72,77 @@ func TestIsEnabled(t *testing.T) {
...
+			assert.Equal(t, tt.expectTrue, NewComponent().MonitorOverrides(ctx))
+		})
+	}
+}
+
+// test preinstall for component class
+func TestPreInstallcomponent(t *testing.T) {
+	c := fake.NewClientBuilder().Build()
+	ctx := spi.NewFakeContext(c, &vzapi.Verrazzano{}, nil, true)
+	assert.Nil(t, NewComponent().PreInstall(ctx))
```
Comment (anajosep) [inline platform-operator/controllers/verrazzano/component/prometheus/adapter/promadapter_component_test.go:147]: Test for both error (failure scenarios) where an expected error should be returned and no error scenarios (happy path)

File: platform-operator/controllers/verrazzano/component/prometheus/adapter/promadapter_test.go [hunk 4]
```diff
@@ -112,3 +113,82 @@ func TestIsPrometheusAdapterReady(t *testing.T) {
...
+			},
+			oV1Alpha1.ValueOverrides,
+		},
+		{
+			"Empty overrides when component nil",
+			&v1beta1.Verrazzano{},
+			[]v1beta1.Overrides{},
+		},
+		{
+			"overrides when component not nil",
```
Comment (anajosep) [inline platform-operator/controllers/verrazzano/component/prometheus/adapter/promadapter_test.go:174]: Add GIVEN/WHEN/THEN style comments.

File: platform-operator/controllers/verrazzano/component/prometheus/kubestatemetrics/kubestatemetrics_component_test.go [hunk 5]
```diff
@@ -71,3 +74,89 @@ func TestIsEnabled(t *testing.T) {
...
+	falseValue := false
+	trueValue := true
+	tests := []struct {
+		name       string
+		actualCR   vzapi.Verrazzano
+		expectTrue bool
+	}{
+		{
+			// GIVEN a default Verrazzano custom resource
+			// WHEN we call IsReady on the KubeStateMetricsComponent
```
Comment (anajosep) [inline platform-operator/controllers/verrazzano/component/prometheus/kubestatemetrics/kubestatemetrics_component_test.go:96]: Correct the method name in the comment

File: platform-operator/controllers/verrazzano/component/prometheus/adapter/promadapter_component_test.go (hunk 2 above)
Comment (rahulsrivastava71) [reply platform-operator/controllers/verrazzano/component/prometheus/adapter/promadapter_component_test.go:80]: in this method i have written two functions for preinstall where dry run is once false and true and this is checking for NotNil which is correct condition

Comment (rahulsrivastava71) [reply platform-operator/controllers/verrazzano/component/prometheus/adapter/promadapter_component_test.go:80]: this is just checking isReady component and its unit and rest other functionality is checked in other test case from it is called inside

File: platform-operator/controllers/verrazzano/component/prometheus/adapter/promadapter_component_test.go (hunk 3 above)
Comment (rahulsrivastava71) [reply platform-operator/controllers/verrazzano/component/prometheus/adapter/promadapter_component_test.go:147]: this functions return err in return and the positive scenario is checked in TestIsPrometheusAdapterReady

File: platform-operator/controllers/verrazzano/component/prometheus/adapter/promadapter_test.go (hunk 4 above)
Comment (rahulsrivastava71) [reply platform-operator/controllers/verrazzano/component/prometheus/adapter/promadapter_test.go:174]: there were test cases in which it was not given so i didn't gave here , if you still wants i can provide

File: platform-operator/controllers/verrazzano/component/prometheus/kubestatemetrics/kubestatemetrics_component_test.go (hunk 5 above)
Comment (rahulsrivastava71) [reply platform-operator/controllers/verrazzano/component/prometheus/kubestatemetrics/kubestatemetrics_component_test.go:96]: ok

File: platform-operator/controllers/verrazzano/component/prometheus/adapter/promadapter_component_test.go (hunk 1 above)
Comment (rahulsrivastava71) [reply platform-operator/controllers/verrazzano/component/prometheus/adapter/promadapter_component_test.go:119]: done

**Review by anajosep — APPROVED** [review]

**Pull request merged**